internal/gatewayapi/contexts.go

```diff
@@ -527,40 +209,95 @@ type TCPRouteContext struct {
 
 	*v1alpha2.TCPRoute
 
-	parentRefs map[v1beta1.ParentReference]*RouteParentContext
+	ParentRefs map[v1beta1.ParentReference]*RouteParentContext
+}
+
+// GetRouteType returns the Kind of the Route object, HTTPRoute,
+// TLSRoute, TCPRoute, UDPRoute etc.
+func GetRouteType(route RouteContext) v1beta1.Kind {
+	rv := reflect.ValueOf(route).Elem()
+	return v1beta1.Kind(rv.FieldByName("Kind").String())
 }
 
-func (t *TCPRouteContext) GetRouteType() v1beta1.Kind {
-	return KindTCPRoute
+// TODO: [v1alpha2-v1beta1] This should not be required once all Route
+// objects being implemented are of type v1beta1.
+// GetHostnames returns the hosts targeted by the Route object.
+func GetHostnames(route RouteContext) []string {
+	rv := reflect.ValueOf(route).Elem()
+	rt := rv.FieldByName("Kind").String()
+	if strings.Contains(rt, "TCP") || strings.Contains(rt, "UDP") {
```

Comment:
can we use existing constants here https://github.com/envoyproxy/gateway/blob/ca36e3f0b11f9d742796f136ad0853295b37352d/internal/gatewayapi/translator.go#L20

Reply:
sure, i'm missing that details